contracts/controller/DAOToken.sol

```diff
@@ -17,12 +17,29 @@ contract DAOToken is ERC827Token,MintableToken,BurnableToken {
     string public name;
     string public symbol;
     uint public constant DECIMAL = 18;
+    uint public cap;
 
     /**
-     * @dev the constructor takes a token name and a symbol
-     */
-    function DAOToken(string _name, string _symbol) public {
+    * @dev Constructor
+    * @param _name - token name
+    * @param _symbol - token symbol
+    * @param _cap - token cap - 0 value means no cap
+    */
+    function DAOToken(string _name, string _symbol,uint _cap) public {
         name = _name;
         symbol = _symbol;
+        cap = _cap;
+    }
+
+    /**
+     * @dev Function to mint tokens
+     * @param _to The address that will receive the minted tokens.
+     * @param _amount The amount of tokens to mint.
+     * @return A boolean that indicates if the operation was successful.
+     */
+    function mint(address _to, uint256 _amount) onlyOwner canMint public returns (bool) {
+        if (cap > 0)
+            require(totalSupply_.add(_amount) <= cap);
```

Comment:
I do not have a strong opinion re require vs return here  though I feel that require is more suitable here as it will return the remaining gas back.

Reply:
Revert as well.
I prefer return, as it keeps the same structure as the regular mint function.

Reply:
If you return ftom a function it consumes all thr gas?

Reply:
@leviadam though the regular mint function return boolean , it will return true for success and will revert otherwise.

giving another thought on that , return false (without revert) here will require the callers of this function to always check its return value ...though it is a good practice to check the return value of each func call and I guess there are static code analysis tools which can detect that it might require additional attention while reviewing the code.
also we might end up with additional gas cost overhead overall for checking the return value.

For example ,here is a point which will need to be modified to check the return value:
https://github.com/daostack/arc/blob/master/contracts/universalSchemes/VestingScheme.sol#L241

Reply:
@dev-matan-tsuberi If you return from a function it consumes the gas used till that point.(GasLimit - GasConsumed).same as revert .

Reply:
OK. I think this is a good solution.